EMS/client-helper-bundle/src/Command/Local/UploadAssetsCommand.php

```diff
@@ -62,23 +65,18 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         }
 
         try {
-            $assetsArchive = $this->localHelper->makeAssetsArchives($baseUrl);
-        } catch (\Throwable $e) {
-            $this->io->error($e->getMessage());
-
-            return self::EXECUTE_ERROR;
-        }
-
-        try {
-            $progressBar = $this->io->createProgressBar($assetsArchive->getSize());
-            $hash = $this->coreApi->file()->uploadFile(
-                $assetsArchive->path,
-                'application/zip',
-                'bundle.zip',
-                fn (string $chunk) => $progressBar->advance(\strlen($chunk))
-            );
-
-            $progressBar->finish();
+            switch ($this->archiveType) {
+                case self::ARCHIVE_ZIP:
+                    $hash = $this->uploadZipArchive();
+                    break;
+                case self::ARCHIVE_EMS:
+                    $hash = $this->uploadEmsArchive();
+                    break;
+                default:
+                    $this->io->error(\sprintf('Archive format %s not supported. Supported formats are "%s" and "%s"', $this->archiveType, self::ARCHIVE_EMS, self::ARCHIVE_ZIP));
+
+                    return self::EXECUTE_ERROR;
+            }
```

Comment:
Ik like match over switch:

```suggestion
            $hash = match ($this->archiveType) {
                  self::ARCHIVE_ZIP => $this->uploadZipArchive(),
                  self::ARCHIVE_EMS => $this->uploadEmsArchive(),
                  default => false
            }()
            
            if (!$hash) {
                  $this->io->error(\sprintf('Archive format %s not supported. Supported formats are "%s" and "%s"', $this->archiveType, self::ARCHIVE_EMS, self::ARCHIVE_ZIP));

                    return self::EXECUTE_ERROR;
            }
};
```